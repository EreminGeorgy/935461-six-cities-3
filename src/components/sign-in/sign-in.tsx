import * as React from "react";
import {useRef, useCallback} from "react";
import PropTypes from "prop-types";
import Header from "../header/header.jsx";
import {connect} from "react-redux";
import {getSelectedCity} from "../../reducer/data/selectors";

export const SignIn = (props) => {
  const {city, onSubmit} = props;

  if (!city) {
    return <p>No data loaded</p>;
  }

  const mailRef = useRef(null);
  const passwordRef = useRef(null);

  const handleSubmit = (evt) => {

    evt.preventDefault();

    onSubmit({
      login: mailRef.current.value,
      password: passwordRef.current.value,
    });
  };

  const memoSubmit = useCallback(handleSubmit, []);

  return (
    <div className="page page--gray page--login">
      <Header />
      <main className="page__main page__main--login">
        <div className="page__login-container container">
          <section className="login">
            <h1 className="login__title">Sign in</h1>
            <form className="login__form form" action="#" method="post" onSubmit={memoSubmit}>
              <div className="login__input-wrapper form__input-wrapper">
                <label className="visually-hidden">E-mail</label>
                <input className="login__input form__input" type="email" name="email" placeholder="Email" required="" ref={mailRef}/>
              </div>
              <div className="login__input-wrapper form__input-wrapper">
                <label className="visually-hidden">Password</label>
                <input className="login__input form__input" type="password" name="password" placeholder="Password" required="" ref={passwordRef}/>
              </div>
              <button className="login__submit form__submit button" type="submit">Sign in</button>
            </form>
          </section>
          <section className="locations locations--login locations--current">
            <div className="locations__item">
              <a className="locations__item-link" href="#">
                <span>
                  {city.name}
                </span>
              </a>
            </div>
          </section>
        </div>
      </main>
    </div>
  );
};

SignIn.propTypes = {
  city: PropTypes.shape({
    name: PropTypes.string,
  }),
  onSubmit: PropTypes.func,
};

const mapStateToProps = (state) => ({
  city: getSelectedCity(state),
});

export default connect(mapStateToProps)(SignIn);
