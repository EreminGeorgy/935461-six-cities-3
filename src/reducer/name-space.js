const NameSpace = {
  DATA: `DATA`,
  APPLICATION: `APPLICATION`,
  USER: `USER`,
  COMMENTS: `COMMENTS`,
  FAVORITES: `FAVORITES`,
};

export default NameSpace;
