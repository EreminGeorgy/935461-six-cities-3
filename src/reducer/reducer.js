import {offers} from '../mocks/offers';
import {extend, getCities, getOffersByCity} from '../utils/utils.js';

const citiesList = getCities(offers);
const initialOffers = getOffersByCity(offers, citiesList[0]);

const initialState = {
  activeCity: citiesList[0],
  offersInActiveCity: initialOffers,
  cities: citiesList,
};

const ActionCreator = {
  newCity: (newCity) => ({
    type: `NEW_CITY`,
    payload: newCity,
  }),
  getOffers: (newCity) => ({
    type: `GET_OFFERS`,
    payload: newCity,
  }),
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case `NEW_CITY`:
      return extend(state, {
        activeCity: action.payload,
      });
    case `GET_OFFERS`:
      return extend(state, {
        offersInActiveCity: getOffersByCity(offers, action.payload),
      });
  }
  return state;
};

export {reducer, ActionCreator};
