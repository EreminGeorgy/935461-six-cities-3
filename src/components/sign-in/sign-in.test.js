import React from "react";
import renderer from "react-test-renderer";
import {SignIn} from "./sign-in.jsx";
import reducer from '../../reducer/reducer.js';
import {createStore} from 'redux';
import {Provider} from 'react-redux';

const store = createStore(
    reducer
);

it(`<SignIn /> should render SignIn`, () => {
  const tree = renderer
    .create(
        <Provider store={store}>
          <SignIn/>
        </Provider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
