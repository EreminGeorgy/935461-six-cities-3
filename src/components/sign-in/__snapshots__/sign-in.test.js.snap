// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignIn /> should render SignIn 1`] = `
<p>
  No data loaded
</p>
`;
